server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java

```diff
@@ -64,6 +66,11 @@ public NodeInfo(StreamInput in) throws IOException {
         } else {
             transportVersion = TransportVersion.fromId(version.id);
         }
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_500_075)) {
+            indexVersion = IndexVersion.readVersion(in);
+        } else {
+            indexVersion = IndexVersion.fromId(version.id);
```

Comment:
This is ok without any further checks. All released versions <8.11 have version-indexversion bijections, and serverless will quickly be updated to versions that know about this. The worst case is that the reported index version is incorrect, but this value isn't actually used anywhere.